3-limit-service-time/main.go

```diff
@@ -21,10 +26,22 @@ type User struct {
 // HandleRequest runs the processes requested by users. Returns false
 // if process had to be killed
 func HandleRequest(process func(), u *User) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second * 10)
+	defer cancel()
+	
 	process()
-	return true
+	
+	return func() bool {
+		select{
+		case <- ctx.Done():
+			return false
+
+		default:
+			return true
+		}
+	} ()
```

Comment:
The intent was to kill the `process()` to prevent freemium users from clogging the resources. This returns correct value, but the RunMockServer takes 15 seconds, instead 14 seconds, indicating that the user-provided process will run as long as it wants.